### PR TITLE
Supporting interruptible for map tasks

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
Implemented IsInterruptible on SubTaskExecutionMetadata. This is set by tracking the number of system failures for each subtask and comparing to the InterruptibleFailureThreshold the same way this is performed in FlytePropeller for individual tasks.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
^^^

Fixing this by overriding the TaskExecutionContext IsInterruptible function means that we can essentially revert the [previous commit](https://github.com/flyteorg/flyteplugins/pull/214) adding an omitInterruptible flag which was used to disable support for interruptible nodes within map tasks.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1533

## Follow-up issue
_NA_

### CODE REVIEW

File: go/tasks/plugins/array/k8s/management.go [hunk 1]
```diff
@@ -107,6 +107,26 @@ func LaunchAndCheckSubTasksState(ctx context.Context, tCtx core.TaskExecutionCon
...
+	// If the current State is newly minted then we must initialize SystemFailures to track how many
+	// times the subtask failed due to system issues, this is necessary to correctly evaluate
+	// interruptible subtasks.
+	if len(currentState.SystemFailures.GetItems()) == 0 {
+		count := uint(currentState.GetExecutionArraySize())
+		maxValue := bitarray.Item(tCtx.TaskExecutionMetadata().GetInterruptibleFailureThreshold())
+
+		systemFailuresArray, err := bitarray.NewCompactArray(count, maxValue)
+		if err != nil {
+			logger.Errorf(context.Background(), "Failed to create system failures array with [count: %v, maxValue: %v]", count, maxValue)
```
[the remark this answers is not in the document]
Comment (EngHabu) [reply go/tasks/plugins/array/k8s/management.go:119]: If this is an error, why are we not returning an error?

[the remark this answers is not in the document]
Comment (hamersaw) [reply go/tasks/plugins/array/k8s/management.go:119]: Probably should be, thanks. I updated it.

**Review by EngHabu — APPROVED** [review]
